src/modules/billing/controllers/batches.js

```diff
@@ -174,13 +173,21 @@ const getBatchDownloadData = async request => {
   const invoices = await invoiceService.getInvoicesForBatchDownload(batch)
 
   // Create a new set to remove duplicate values
-  const chargeVersionIds = [...new Set(flatMap(invoices.map(invoice => {
-    return flatMap(invoice.billingInvoiceLicences.map(invoiceLicence =>
-      invoiceLicence.billingTransactions
-        .filter(transaction => !!transaction.chargeElement)
-        .map(transaction => transaction.chargeElement.chargeVersionId)
-    ))
-  })))]
+  // Define an array of unique charge version IDS
+  const chargeVersionIds = [...new Set(
+    // Map over the invoice array and return an array of charge version IDs
+    invoices.map(invoice => {
+      // Get an array of billing invoice licences and filter out transactions
+      const filtered = invoice.billingInvoiceLicences.map(invoiceLicence =>
+        invoiceLicence.billingTransactions
+          .filter(transaction => !!transaction.chargeElement)
+          .map(transaction => transaction.chargeElement.chargeVersionId)
+      )
+      // Return a flattened array of charge version IDs
+      return filtered.flatMap(n => n)
+    }).flatMap(n => n)
```

Comment:
If we replace this with `.flat(Infinity)` then we don't need the `.flatMap()` on line 187 as it will completely flatten the resulting array. But it might be worth replacing some or all of the other `.map()` in this with `.flatMap()`, which would hopefully give us a flat array without needing an extra 'and now flatten what we have' step.